lib/jsdom/living/file-reader.js

```diff
@@ -0,0 +1,159 @@
+"use strict";
+
+const DOMException = require("../web-idl/DOMException");
+const EventTarget = require("./generated/events/EventTarget");
+
+const blobSymbols = require("./blob-symbols");
+
+function FileReaderEventTarget() {
+  if (!(this instanceof FileReaderEventTarget)) {
+    throw new TypeError("DOM object constructor cannot be called as a function.");
+  }
+  EventTarget.setup(this);
+}
+
+FileReaderEventTarget.prototype = Object.create(EventTarget.interface.prototype);
+
+module.exports = function (core) {
+  core.createFileReader = function createFileReader(window) {
+    const ProgressEvent = core.ProgressEvent;
+
+    class FileReader extends FileReaderEventTarget {
+      constructor() {
+        super();
+        this.error = null;
+        this.readyState = FileReader.EMPTY;
+        this.result = null;
+        this.onloadstart = null;
+        this.onprogress = null;
+        this.onload = null;
+        this.onabort = null;
+        this.onerror = null;
+        this.onloadend = null;
+      }
+      get EMPTY() {
```

Comment:
Constants should not be getters; use addConstants to add them (even for classes).